src/sentry/models/organization.py

```diff
@@ -196,6 +196,9 @@ def save(self, *args, **kwargs):
             with TimedRetryPolicy(10)(lock.acquire):
                 slugify_target = slugify_target.replace("_", "-").strip("-")
                 slugify_instance(self, slugify_target, reserved=RESERVED_ORGANIZATION_SLUGS)
+                if self.slug.endswith("-"):
```

Comment:
If a trailing `-` isn't valid should `slugify_instance()` take care of that? Having to strip and re-generate is an awkward interface.